app/screens/send-bitcoin-screen/send-bitcoin-destination-screen.tsx

```diff
@@ -65,6 +66,33 @@ gql`
   }
 `
 
+type Wallets =
+  | readonly (
+      | {
+          readonly __typename: "BTCWallet"
+          readonly id: string
+        }
+      | {
+          readonly __typename: "UsdWallet"
+          readonly id: string
+        }
+    )[]
+  | undefined
+
+type Contacts =
+  | readonly {
+      readonly __typename: "UserContact"
+      readonly id: string
+      readonly username: string
+    }[]
+  | undefined
+
+type NetworkWalletsContacts = {
+  _bitcoinNetwork: Network | undefined
+  _wallets: Wallets
+  _contacts: Contacts
```

Comment:
network, wallets and contacts are immutable value in the context of this flow. do we need to move those values around? It's not clear to me why we would want to do that or what would be the benefits of it

Reply:
* immutable as is: they won't change, unless if there were out of sync and will be re-sync with the query to the backend server when entering the screen.

and really contacts it the only one that could be updated following a payment. those other values are value that won't change for a particular user, so they would only change after a log out / login with new user or login to new environment.

Reply:
Didn't know they were immutable. Saw them coming in from GraphQL queries, and they weren't really being handled as such in the rest of the code. If they're immutable though you're right I could remove them from the setTimeout -> validateDestination functions and greatly simplify the code.